src/VisualStudio/IntegrationTest/TestUtilities/InProcess2/InProcComponent2.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Threading;
+using EnvDTE;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess2
+{
+    public abstract class InProcComponent2
```

Comment:
:memo: This and derived would eventually replace the existing in-process and out-of-process components once the entire test suite is ported.

Reply:
Part of the reason for the existing in-proc/out-of-proc split was to more closely simulate RPS/DDRITs so that we could eventually simulate those numbers ourselves as part of our CI system.